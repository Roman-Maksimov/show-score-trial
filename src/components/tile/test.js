import React from 'react';
import _ from 'lodash';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Tile from './';


const defaultProps = {
  onClick: _.noop,
};

test('should render empty tile', () => {
  const component = renderer.create(
    <Tile {...defaultProps} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render black tile', () => {
  const component = renderer.create(
    <Tile {...defaultProps} isSet />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render blue tile', () => {
  const component = renderer.create(
    <Tile {...defaultProps} isWater />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should write isError to state', () => {
  const component = shallow(
    <Tile {...defaultProps} />,
  );

  component.setProps({ isError: true }, () => {
    expect(component.state().isError).toEqual(true);
  })
});
