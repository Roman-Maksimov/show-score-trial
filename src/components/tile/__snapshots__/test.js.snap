// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render black tile 1`] = `
<div
  className="tile"
  onClick={[Function]}
>
  <div
    className="bgBlack"
  />
</div>
`;

exports[`should render blue tile 1`] = `
<div
  className="tile"
  onClick={[Function]}
>
  <div
    className="bgBlue"
  />
</div>
`;

exports[`should render empty tile 1`] = `
<div
  className="tile"
  onClick={[Function]}
>
  <div
    className="bg"
  />
</div>
`;
